tests/infrastructure/sqlalechemy/repositories/test_base_repository.py

```diff
@@ -0,0 +1,198 @@
+        mock_object = MockDomainModel(payload={'endboss': 'Melkor'})
+        job_id = uuid4()
+
+        with pytest.raises(EntityNotFoundError):
+            MockBaseRepository(db_connection=transactional_session).get_by_job_id(job_id)
+
+
+class TestBaseRepositoryUpdate:
+
+    def test_if_entity_in_database_then_return_updated_domain_representation(self, transactional_session):
+        mock_object = MockDomainModel(payload={'endboss': 'Melkor'})
+        schema_object = MockSchema(**mock_object.dict())
+        transactional_session.add(schema_object)
+        transactional_session.commit()
+        updated_object = MockDomainModel(payload={'endboss': 'Sauron'})
+
+        updated_entity = MockBaseRepository(db_connection=transactional_session).update(schema_object.id, updated_object)
+
+        assert updated_entity == updated_object
+
+    def test_if_entity_in_database_then_update_in_database(self, transactional_session):
+        mock_object = MockDomainModel(payload={'endboss': 'Melkor'})
+        schema_object = MockSchema(**mock_object.dict())
+        transactional_session.add(schema_object)
+        transactional_session.commit()
+        updated_object = MockDomainModel(payload={'endboss': 'Sauron'})
+
+        MockBaseRepository(db_connection=transactional_session).update(schema_object.id, updated_object)
+
+        updated_entry_in_db = transactional_session.query(MockSchema).filter(MockSchema.id == schema_object.id).first()
+        assert updated_entry_in_db.id == schema_object.id
+        assert updated_entry_in_db.payload == {'endboss': 'Sauron'}
+
+    def test_if_entity_not_in_database_then_raise_error(self, transactional_session):
+        mock_object = MockDomainModel(payload={'endboss': 'Melkor'})
+        schema_object = MockSchema(**mock_object.dict())
+        updated_object = MockDomainModel(payload={'endboss': 'Sauron'})
+
+        with pytest.raises(EntityNotFoundError):
+            MockBaseRepository(db_connection=transactional_session).update(schema_object.id, updated_object)
+
+
+class TestBaseRepositoryDelete:
+
+    def test_if_entity_in_database_then_delete_from_database(self, transactional_session):
+        mock_object = MockDomainModel(payload={'endboss': 'Melkor'})
+        schema_object = MockSchema(**mock_object.dict())
+        transactional_session.add(schema_object)
+        transactional_session.commit()
+
+        MockBaseRepository(db_connection=transactional_session).delete(schema_object.id)
+
+        assert len(transactional_session.query(MockSchema).all()) == 0
+
+    def test_if_entity_not_in_database_then_raise_error(self, transactional_session):
+        mock_object = MockDomainModel(payload={'endboss': 'Melkor'})
+        schema_object = MockSchema(**mock_object.dict())
+
+        with pytest.raises(EntityNotFoundError):
+            MockBaseRepository(db_connection=transactional_session).delete(schema_object.id)
+
```

Comment:
Wouldn't it be enough to test the erica repository?

Reply:
Yes, we could also do that. However, as the implementation is in the base repository, imo it made more sense to test the base repository and then only add tests for special implementation of the erica repository